v20.2/take-and-restore-encrypted-backups.md

```diff
@@ -6,47 +6,145 @@ toc: true
+
+On [`RESTORE`](#restore-from-an-encrypted-backup-with-aws-kms), CockroachDB reads the `ENCRYPTION_INFO` file and attempts to decrypt the encrypted data key using the KMS URI provided in the `RESTORE` statement. Once CockroachDB successfully obtains the unencrypted data key, the `BACKUP` manifest and data files will be decrypted and the restoration will proceed. Similarly, the same KMS URI is needed to decrypt the file to list the contents of the backup when using [`SHOW BACKUP`](show-backup.html#show-an-encrypted-backup).
+
+When used with [incremental backups](backup.html#incremental-backups), the `kms` option is applied to all the [backup file URLs](backup.html#backup-file-urls), which means each incremental must include at least one of the KMS URIs used to take the full backup. It can be any subset of the original URIs, but you cannot include any new KMS URIs. Similarly, when used with [locality-aware backups](take-and-restore-locality-aware-backups.html), the KMS URI provided is applied to files in all localities.
+
+For more information about AWS KMS, see the [documentation](https://aws.amazon.com/kms/).
+
+### Generate a customer master key (CMK)
+
+Before you can use AWS KMS to encrypt a CockroachDB backup, you must first generate a **customer master key (CMK)**. This is the master key generated by AWS KMS and it never leaves the KMS. It contains key-related metadata and key material to encrypt/decrypt other data. The key material can never be exported, deleted, or extracted. CockroachDB expects the CMK to be symmetric (256 bit).
+
+<a name="multi-region"></a> There is also the option to [use multi-region encryption on your data](#take-a-backup-with-multi-region-encryption). At the time of `BACKUP`, you can [provide multiple AWS KMS URIs](#aws-kms-uri-format), each referencing a CMK in a different AWS region. This allows CockroachDB to save multiple versions of the encrypted data key used to encrypt the backup data, one per AWS KMS URI. With these encrypted versions of the data key stored alongside the encrypted backup data, a user can `RESTORE` the encrypted data using any one of the KMS URIs that were supplied during backup. In the case of a single KMS region outage, the data can be decrypted with any of the CMKs from the other regions.
+
+### AWS KMS URI format
+
+The AWS KMS URI must use the following format:
+
+~~~
+aws:///<host>?AUTH=<auth_type>&REGION=<region>
+~~~
+
+The AWS URI **requires** the following:
 
-## Create an encrypted backup
+ Component                  | Description
+----------------------------+------------------------------------------------------------------------
+`aws:///`                   | The AWS scheme. Note the triple slash (`///`).
+`<host>`                    | The key identifiers used to reference the CMK that should be used to encrypt or decrypt. For information about the supported formats, see the [AWS KMS docs](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id).
```

Comment:
@lnhsingh can we switch the component name from `<host>` to `cmk` or something along those lines? The term host conflicts with the URI semantics of the term "host" which refers to what is specified after the second forward slash in a URI (egs: aws://host/path). Since <host> is not actually specified in the "host" slot of the URI it might lead to confusion!

Reply:
Addressed with https://github.com/cockroachdb/docs/pull/8775